### PR TITLE
Add oauth

### PR DESCRIPTION
**Pull request by booshington**

### CODE REVIEW

**Comment by booshington** [comment]
@khultman Code complete, needs end to end testing from resources currently tied up. Code implemented marked as experimental and examples changed so as to avoid users locking themselves out of their accounts.

Will revisit in later branch upon remediation of previously identified issues, further api refactor.

File: gremlinapi/oauth.py [hunk 1]
```diff
@@ -23,49 +27,285 @@
 
 from http.client import HTTPResponse
```
Comment (khultman) [inline gremlinapi/oauth.py:28]: Exposing http.client outside of the API call class should be avoided. Ideally the API client class should not be returning anything but objects we create.

File: gremlinapi/util.py [hunk 2]
```diff
@@ -17,10 +17,29 @@ def get_version():
 string_types = (type(b""), type(""), type(f""))
 
 
+def experimental(func):
```
Comment (khultman) [inline gremlinapi/util.py:20]: Might be nice to take in an optional message that enhance context on why a method is listed as experimental.

**Review by khultman — CHANGES_REQUESTED** [review]
See comments, also let's minor version update this as it's new functionality.

File: gremlinapi/gremlinapi.py [hunk 3]
```diff
@@ -5,6 +5,7 @@
 import json
 import logging
 import time
+from http.client import HTTPResponse
```
[the remark this answers is not in the document]
Comment (booshington) [reply gremlinapi/gremlinapi.py:8]: The HTTPResponse is a return value that isn't necessary, so removing it is straightforward and removes this dependency.

File: gremlinapi/gremlinapi.py [hunk 4]
```diff
@@ -179,3 +180,39 @@ def _warn_if_not_param(
             log.warning(error_msg)
             param = default
         return param
+
+    @classmethod
+    def auth_toggles(
```
[the remark this answers is not in the document]
Comment (booshington) [reply gremlinapi/gremlinapi.py:185]: Agreed. Code moved.

File: gremlinapi/oauth.py [hunk 5]
```diff
@@ -3,6 +3,10 @@
 # Copyright (C) 2020 Kyle Bouchard <kyle.bouchard@gremlin.com>, Gremlin Inc <sales@gremlin.com>
 
 import logging
+import requests
```
[the remark this answers is not in the document]
Comment (booshington) [reply gremlinapi/oauth.py:6]: Done, refactored to use gremlin http client

File: gremlinapi/util.py (hunk 2 above)
Comment (booshington) [reply gremlinapi/util.py:20]: Agreed. Done.

**Review by khultman — APPROVED** [review]

**Pull request merged**